lib/utils/groupErrors.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+var chalk = require('chalk');
+
+var groupedErrors = {};
+
+var getFinalGroupErrors = function(errorGroup) {
```

Comment:
this function is used only once. and is doing two things in one: is "getting the list of all the errors as a 'kind of' array" and "preparing the formatting of the output messages" (see when you add a double newline at the end). in the same way, `consoleOutputErrArr` is similar. I think it would be better to move these two pieces of code where they are actually used. I understand it may increase the complexity there, but is not that moving the code in another place under a function solves the problem :)
I have tried to refactor the code, removing the extra functions and simplifying the logic. see how it looks in this way:
https://gist.github.com/didoo/2b681bee9e5f1a6a4327d6278e35fa09
(if you want to use it, test it anyway, I've just refactored inside a text editor, not actually tested, ok?)

Reply:
Yeah, before the refactor they were used in multiple places. I thought about redoing as a single func but thought that there were two reasons to keep it. Clarity of use and that if people wanted the other functionality back it would be nice. It seems like you and Danny both are telling me to stop engineering ahead and just focus on immediate task. I'll remove the functions.

Reply:
Yeah, before the refactor they were used in multiple places. I thought about redoing as a single func but thought that there were two reasons to keep it. Clarity of use and that if people wanted the other functionality back it would be nice. It seems like you and Danny both are telling me to stop engineering ahead and just focus on immediate task. I'll remove the functions.

Reply:
Yeah, before the refactor they were used in multiple places. I thought about redoing as a single function but thought that there were two reasons to keep it. Clarity of use and that if people wanted the other functionality back it would be nice. It seems like you and Danny both are telling me to stop engineering ahead and just focus on immediate task. I'll remove the functions.

Reply:
Yeah, before the refactor they were used in multiple places. I thought about redoing as a single function but thought that there were two reasons to keep it. Clarity of use and that if people wanted the other functionality back it would be nice. It seems like you and Danny both are telling me to stop engineering ahead and just focus on immediate task. I'll remove the functions.

Reply:
Yeah, before the refactor they were used in multiple places. I thought about redoing as a single function but thought that there were two reasons to keep it. Clarity of use and that if people wanted the other functionality back it would be nice. It seems like you and Danny both are telling me to stop engineering ahead and just focus on immediate task. I'll remove the functions.